moda-contracts/src/Marketplace.sol

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.21;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "../src/interfaces/IMarketplace.sol";
+import "../src/interfaces/Releases/IReleasesRegistration.sol";
+
+/**
+ * @title Marketplace
+ * @dev This contract allows buying and selling of Releases and charges a fee on each sale.
+ */
+contract Marketplace is IMarketplace, ERC1155Holder, ReentrancyGuard, AccessControl {
+    using SafeERC20 for IERC20;
+
+    // State Variables
+
+    address _usdc;
```

Comment:
Let's change this name to something like "token". An organization might want to use another erc20 token instead of USDC or MODA may want to switch to USDT some day. Its best to keep the contract code as abstract as possible as long as it doesn't complicate things.

Reply:
`IERC20` type